python/phylanx/ast/transformation.py

```diff
@@ -442,15 +446,14 @@ def __init__(self, f):
             assert len(tree.body) == 1
 
             if target == "PhySL":
-                self.__physl_src__ = '%s(%s)\n' % (
-                    full_node_name(tree.body[0], 'block'),
-                    transformation.recompile(tree))
+                self.__physl_src__ = transformation.recompile(tree)
+                et.eval(self.__physl_src__, self.cs)
```

Comment:
Should things be evaluated whenever the decorators is created?

Reply:
@hkaiser It seems to work now. Yes, when the decorator is evaluated, the function is compiled. This is an improvement over what happened before, where the function was compiled on each evaluation. When the function is called, eval is called again, but instead of the function definition, only the function name is needed.

Reply:
Neat trick!

Reply:
OTOH, doesn't this also mean that the function is evaluated twice? Once without arguments (which could btw fail) and once with them? It might be better to explicitly separate compilation and evaluation in the Python interface.